### ISSUE TITLE
Unable to attach a service to a status page

### ISSUE THREAD

**Issue by eelayoubi** [iLert/terraform-provider-ilert#48]
Hello,

We are having an issue adding a service to a status page. Below is the code we are using.

```
terraform {
  required_providers {
    ilert = {
      source  = "iLert/ilert"
      version = "1.11.3"
    }
  }
}

provider "ilert" {
  # Configuration options
  api_token = "THE TOKEN"
}

data "ilert_team" "team" {
  name = "TEAM NAME"
}

resource "ilert_service" "service" {
  name        = "test service 1"
  status      = "OPERATIONAL"
  description = "Test service 1"
  team {
    id = data.ilert_team.team.id
  }
}

resource "ilert_service" "service2" {
  name        = "test service 2"
  status      = "OPERATIONAL"
  description = "Test service 2"
  team {
    id = data.ilert_team.team.id
  }
}

data "ilert_escalation_policy" "escalation_policy" {
  name = "TEST-Escalation"
}

resource "ilert_alert_source" "alert_source" {
  name              = "Test Alert Source"
  integration_type  = "SIGNALFX"
  escalation_policy = data.ilert_escalation_policy.escalation_policy.id
  team {
    id = data.ilert_team.team.id
  }
}

resource "ilert_alert_action" "alert_action" {
  name = "test Alert Action"
  trigger_types = [
    "alert-created",
    "alert-acknowledged",
    "alert-auto-resolved",
    "alert-resolved",
  ]

  alert_source {
    id = ilert_alert_source.alert_source.id
  }

  connector {
    type = "automation_rule"
  }

  automation_rule {
    service_ids    = [ilert_service.service.id]
    service_status = "DEGRADED"
    alert_type     = "CREATED"
  }
}

resource "ilert_status_page" "example" {
  name       = "example status page"
  subdomain  = "example-test.ilert.io"
  visibility = "PRIVATE"
  team {
    id = data.ilert_team.team.id
  }
  service {
    id   = ilert_service.service.id
    name = ilert_service.service.name
  }

  service {
    id   = ilert_service.service2.id
    name = ilert_service.service2.name
  }

  structure {
    element {
      id   = ilert_service.service.id
      type = "SERVICE"
    }
    element {
      id   = ilert_service.service2.id
      type = "SERVICE"
    }
  }

  lifecycle {
    create_before_destroy = true
  }
}
```

Once we run this via terraform apply, everything is created as expected. We have a status page with 2 services attached to it. THen we activate the status page via the console (since we cannot activate it via terraform, which is weird.).

After that, we try to add another service resource, and reference it in the status page resource, here is the updated snippet:

```
resource "ilert_service" "service3" {
  name        = "test service 3"
  status      = "OPERATIONAL"
  description = "Test service 3"
  team {
    id = data.ilert_team.team.id
  }
}


resource "ilert_status_page" "example" {
  name       = "example status page"
  subdomain  = "example-test.ilert.io"
  visibility = "PRIVATE"
  team {
    id = data.ilert_team.team.id
  }
  service {
    id   = ilert_service.service.id
    name = ilert_service.service.name
  }

  service {
    id   = ilert_service.service2.id
    name = ilert_service.service2.name
  }

  service {
    id   = ilert_service.service3.id
    name = ilert_service.service3.name
  }

  structure {
    element {
      id   = ilert_service.service.id
      type = "SERVICE"
    }
    element {
      id   = ilert_service.service2.id
      type = "SERVICE"
    }

    element {
      id   = ilert_service.service3.id
      type = "SERVICE"
    }
  }

  lifecycle {
    create_before_destroy = true
  }
}
```

THen we try running terraform apply, and the whole thing just hangs, I enabled the tracing to see what is happening,
it just fails with the message:

[ERROR] vertex "ilert_status_page.example" error: waiting for status page with id '74015' to be updated

**Comment by krystianity** [comment]
Thanks for opening the issue.

The reason why it is not possible to activate status pages from Terraform is that the action itself is not available through the PUT API, requires an add-on selection (including billing) beforehand and runs a large amount of validations - that are just easier to handle in the UI.

We have identified an issue where the activation state could cause the apply operation to fail in a loop, as the Go client was actually trying to set the activation status back to false (which is a forbidden operation). We have fixed this issue in 1.11.4, please update your provider and try again.

**Comment by eelayoubi** [comment]
Hey @krystianity , thank you for the fix, I did test it and it works fine 🎉
